client/CMT.cpp

```diff
@@ -840,13 +857,88 @@ void dispose()
 	CMessage::dispose();
 }
 
+std::map<uint32_t, std::set<std::pair<int, int>>> getDisplayModes()
+{
+	static const std::map<uint32_t, std::string> formatNames =
+	{
+		{ SDL_PIXELFORMAT_UNKNOWN, "UNKNOWN" },
+		{ SDL_PIXELFORMAT_INDEX1LSB, "INDEX1LSB" },
+		{ SDL_PIXELFORMAT_INDEX1MSB, "INDEX1MSB" },
+		{ SDL_PIXELFORMAT_INDEX4LSB, "INDEX4LSB" },
+		{ SDL_PIXELFORMAT_INDEX4MSB, "INDEX4MSB" },
+		{ SDL_PIXELFORMAT_INDEX8, "INDEX8" },
+		{ SDL_PIXELFORMAT_RGB332, "RGB332" },
+		{ SDL_PIXELFORMAT_RGB444, "RGB444" },
+		{ SDL_PIXELFORMAT_RGB555, "RGB555" },
+		{ SDL_PIXELFORMAT_BGR555, "BGR555" },
+		{ SDL_PIXELFORMAT_ARGB4444, "ARGB4444" },
+		{ SDL_PIXELFORMAT_RGBA4444, "RGBA4444" },
+		{ SDL_PIXELFORMAT_ABGR4444, "ABGR4444" },
+		{ SDL_PIXELFORMAT_BGRA4444, "BGRA4444" },
+		{ SDL_PIXELFORMAT_ARGB1555, "ARGB1555" },
+		{ SDL_PIXELFORMAT_RGBA5551, "RGBA5551" },
+		{ SDL_PIXELFORMAT_ABGR1555, "ABGR1555" },
+		{ SDL_PIXELFORMAT_BGRA5551, "BGRA5551" },
+		{ SDL_PIXELFORMAT_RGB565, "RGB565" },
+		{ SDL_PIXELFORMAT_BGR565, "BGR565" },
+		{ SDL_PIXELFORMAT_RGB24, "RGB24" },
+		{ SDL_PIXELFORMAT_BGR24, "BGR24" },
+		{ SDL_PIXELFORMAT_RGB888, "RGB888" },
+		{ SDL_PIXELFORMAT_RGBX8888, "RGBX8888" },
+		{ SDL_PIXELFORMAT_BGR888, "BGR888" },
+		{ SDL_PIXELFORMAT_BGRX8888, "BGRX8888" },
+		{ SDL_PIXELFORMAT_ARGB8888, "ARGB8888" },
+		{ SDL_PIXELFORMAT_RGBA8888, "RGBA8888" },
+		{ SDL_PIXELFORMAT_ABGR8888, "ABGR8888" },
+		{ SDL_PIXELFORMAT_BGRA8888, "BGRA8888" },
+		{ SDL_PIXELFORMAT_ARGB2101010, "ARGB2101010" },
+		{ SDL_PIXELFORMAT_YV12, "YV12" },
+		{ SDL_PIXELFORMAT_IYUV, "IYUV" },
+		{ SDL_PIXELFORMAT_YUY2, "YUY2" },
+		{ SDL_PIXELFORMAT_UYVY, "UYVY" },
+		{ SDL_PIXELFORMAT_YVYU, "YVYU" }
+	};
+	int displays = SDL_GetNumVideoDisplays();
+	std::map<std::tuple<uint32_t, int, int>, int> modeCounts;
+	for(int display = 0; display < displays; display++)
+	{
+		int modes = SDL_GetNumDisplayModes(display);
+		for(int mode = 0; mode < modes; mode++)
+		{
+			SDL_DisplayMode sdldm;
+			SDL_GetDisplayMode(display, mode, &sdldm);
+			modeCounts[std::make_tuple(sdldm.format, sdldm.w, sdldm.h)]++;
+		}

```

Comment:
May be. But this also makes impossible to use better mode on display 0 if  display 1 does not support it. It is better to fallback to lower resolution only when window has been actually moved to another display.